web/models/types/entity_type.ex

```diff
@@ -0,0 +1,27 @@
+defmodule CoursePlanner.Types.EntityType do
+  @moduledoc """
+    This module introduces a custom type for Etco for checking status in the model
+  """
+  @behaviour Ecto.Type
+  def type, do: :string
+
+  @valid_entity_types ['Planned', 'Active', 'Finished', 'Graduated', 'Frozen', 'Deleted']
+
+  def valid_entity_types, do: @valid_entity_types
```

Comment:
fyi, he explained to me the idea of this function is to provide possible values to the views